### PR TITLE
redact auth provider secrets in UI

### PR DESCRIPTION
**Pull request by mollylogue**
Under the site configuration page in site-admin, the secrets in the config should display as "REDACTED" rather than the actual secret. I've currently only redacted for open id, github, gitlab, and the executors.accessToken field because those were the obvious ones that jumped out to me, but we can redact others as well if needed. 

Closes https://github.com/sourcegraph/sourcegraph/issues/28940

### CODE REVIEW

**Review by asdine — APPROVED** [review]

**Review by ryanslade — APPROVED** [review]
LGTM assuming you've also done a manual test that includes editing a redacted value?

File: cmd/frontend/graphqlbackend/site.go [hunk 1]
```diff
@@ -175,7 +175,11 @@ func (r *schemaResolver) UpdateSiteConfiguration(ctx context.Context, args *stru
 	}
 
 	prev := globals.ConfigurationServerFrontendOnly.Raw()
-	prev.Site = args.Input
+	unredacted, err := conf.UnredactSecrets(args.Input, prev)
+	if err != nil {
+		return false, errors.Errorf("error unredacting secrets: %s", unredacted)
```
[the remark this answers is not in the document]
Comment (mollylogue) [reply cmd/frontend/graphqlbackend/site.go:180]: Oh yes definitely meant to print `err` instead 🤦‍♀️

**Comment by mollylogue** [comment]
> LGTM assuming you've also done a manual test that includes editing a redacted value?

Yep! I've tested that. It successfully updates and the secret is displayed as `"REDACTED"` when the page is reloaded.

File: cmd/frontend/graphqlbackend/site.go [hunk 2]
```diff
@@ -140,8 +140,8 @@ func (r *siteConfigurationResolver) EffectiveContents(ctx context.Context) (JSON
 	if err := backend.CheckCurrentUserIsSiteAdmin(ctx, r.db); err != nil {
 		return "", err
 	}
-	siteConfig := globals.ConfigurationServerFrontendOnly.Raw().Site
-	return JSONCString(siteConfig), nil
+	siteConfig, err := conf.RedactSecrets(globals.ConfigurationServerFrontendOnly.Raw())
+	return JSONCString(siteConfig.Site), err
```
[the remark this answers is not in the document]
Comment (mollylogue) [reply cmd/frontend/graphqlbackend/site.go:144]: 👍

**Pull request merged**